File: apps/chat/libs/chat-domain/src/main/java/com/example/chat/domain/message/MessageId.java
package com.example.chat.domain.message;

import java.util.UUID;

/**
 * 메시지 ID (Value Object)
 */
public record MessageId(String value) {
    public MessageId {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("MessageId cannot be null or blank");
        }
    }

    public static MessageId of(String value) {
        return new MessageId(value);
    }

    public static MessageId generate() {
        return new MessageId(UUID.randomUUID().toString());
    }

    public String getValue() {
        return value;
    }
}

File: apps/chat/libs/chat-domain/src/main/kotlin/com/example/chat/domain/message/MessageId.kt
package com.example.chat.domain.message

import java.util.UUID

/**
 * 메시지 ID (Value Object)
 */
data class MessageId(
    val value: String
) {
    init {
        require(value.isNotBlank()) { "MessageId cannot be null or blank" }
    }

    companion object {
        @JvmStatic
        fun of(value: String): MessageId = MessageId(value)

        @JvmStatic
        fun generate(): MessageId = MessageId(UUID.randomUUID().toString())
    }
}
